lc-hax/Scripts/Core/HaxCamera.cs

```diff
@@ -0,0 +1,162 @@
+using GameNetcodeStuff;
+using Hax;
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+
+internal class HaxCamera : MonoBehaviour {
+    internal static HaxCamera? Instance { get; private set; }
+    internal KeyboardMovement? KeyboardMovement { get; private set; }
+    internal MousePan? MousePan { get; private set; }
+    internal GameObject? HaxCameraContainer { get; private set; }
+    internal GameObject? HaxCamAudioContainer { get; private set; }
+    internal Camera? CustomCamera { get; private set; }
+    internal AudioListener? HaxCamAudioListener { get; private set; }
+
+    void Awake() {
+        Instance = this;
+        _ = this.GetCamera();
+    }
+
+    void OnEnable() {
+        GameListener.OnGameStart += this.DisableCamera;
+        GameListener.OnGameEnd += this.DisableCamera;
+    }
+
+    void OnDisable() {
+        GameListener.OnGameStart -= this.DisableCamera;
+        GameListener.OnGameEnd -= this.DisableCamera;
+    }
+
+    internal void SetActive(bool active) {
+        if (Helper.StartOfRound is not StartOfRound startOfRound) return;
+        if (Helper.LocalPlayer is not PlayerControllerB player) return;
+        if (player.activeAudioListener is not AudioListener playerlistener) return;
+        if (this.HaxCameraContainer is null) return;
+        if (this.HaxCamAudioListener is not AudioListener haxListener) return;
+        if (this.GetCamera() is not Camera camera) return;
+
+        this.HaxCameraContainer.SetActive(active);
+
+        if (!this.HaxCameraContainer.activeSelf) {
+            playerlistener.enabled = true;
+            startOfRound.audioListener = playerlistener;
+            startOfRound.activeCamera.enabled = true;
+        }
+        else {
+            this.CopyFromCamera(startOfRound.activeCamera.transform, camera, startOfRound.activeCamera);
+            startOfRound.activeCamera.enabled = false;
+            playerlistener.enabled = false;
+            startOfRound.audioListener = haxListener;
+        }
+    }
```

Comment:
Is there a reason why this cannot be in `OnEnable`?

Reply:
Because the container is disabled, not the manager

Reply:
I am aware, but it doesn't seem like there would be a problem if we use `OnEnable` instead. Then we don't have to use the boolean flags.